internal/provider/resource_consul_cluster_test.go

```diff
@@ -14,26 +15,84 @@ import (
 var consulClusterUniqueID = fmt.Sprintf("test-%s", time.Now().Format("200601021504"))
 var consulClusterHVNUniqueID = fmt.Sprintf("test-hvn-%s", time.Now().Format("200601021504"))
 
-var consulCluster = fmt.Sprintf(`
-resource "hcp_consul_cluster" "test" {
-	cluster_id         = "%s"
-	hvn_id             = hcp_hvn.test.hvn_id
-	tier               = "development"
-	min_consul_version = data.hcp_consul_versions.test.recommended
-}
-`, consulClusterUniqueID)
+var consulCluster = consulClusterConfig(consulClusterUniqueID, `
+	ip_allowlist {
+    	address = "172.25.16.0/24"
+    	description = "this is an IPV4 address"
+  	}
+`)
 
-var updatedConsulCluster = fmt.Sprintf(`
-resource "hcp_consul_cluster" "test" {
-	cluster_id = "%s"
-	hvn_id     = hcp_hvn.test.hvn_id
-	tier       = "standard"
+var updatedConsulCluster = consulClusterConfig(consulClusterUniqueID, `
 	size	   = "small"
+
+	ip_allowlist {
+    	address = "172.25.14.0/24"
+ 		description = "this is an updated IPV4 address"
+    }
+`)
+
+var updateConsulClusterDuplicateCIDR = consulClusterConfig(consulClusterUniqueID, `
+	ip_allowlist {
+    	address = "172.25.14.0/24"
+    	description = "this is an IPV4 address"
+	}
+
+	ip_allowlist {
+    	address = "172.25.14.0/24"
+    	description = "this is a duplicate IPV4 address"
+	}
+`)
+
+var createConsulClusterDuplicateCIDR = consulClusterConfig("test-failure", `
+	ip_allowlist {
+    	address = "172.25.16.0/24"
+    	description = "this is an IPV4 address"
+	}
+
+	ip_allowlist {
+    	address = "172.25.16.0/24"
+    	description = "this is a duplicate IPV4 address"
+	}
+`)
+
+var createConsulClusterCIDRExceeded = consulClusterConfig("test-failure", `
+	ip_allowlist {
+    	address = "172.25.16.0/24"
+    	description = "this is a first IPV4 address"
+	}
+	
+	ip_allowlist {
+    	address = "172.25.10.0/24"
+    	description = "this is a secondIPV4 address"
+	}
+
+	ip_allowlist {
+    	address = "172.25.13.0/24"
+    	description = "this is a third IPV4 address"
+	}
+
+	ip_allowlist {
+    	address = "172.25.12.0/24"
+    	description = "this is a fourth IPV4 address, exceeding allowed limit"
+	}
+`)
+
+func consulClusterConfig(clusterID string, opt string) string {
```

Comment:
Created this helper function that creates the `hcp_consul_cluster` for the tests cases.